Parser/lexer/lexer.c

```diff
@@ -1317,15 +1329,18 @@ tok_get_fstring_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct
     tok->multi_line_start = tok->line_start;
     while (end_quote_size != current_tok->f_string_quote_size) {
         int c = tok_nextc(tok);
-        if (tok->done == E_ERROR) {
+        if (tok->done == E_ERROR || tok->done == E_DECODE) {
             return MAKE_TOKEN(ERRORTOKEN);
         }
         int in_format_spec = (
-                current_tok->last_expr_end != -1
+                current_tok->in_format_spec
                 &&
                 INSIDE_FSTRING_EXPR(current_tok)
         );
 
+       if (!_PyLexer_update_fstring_expr(tok, '{')) {
```

Comment:
Not sure if there is a better way around this or if I am missing something. This code is so we get the debug buffers updated when we start the FSTRING MIDDLE in the format part. Check the trees in the other comments to see why we need to capture those